lib/vfs/src/vfs_fs.rs

```diff
@@ -0,0 +1,183 @@
+use crate::*;
+use std::io::{Read, Seek, Write};
+use std::path::Path;
+use std::sync::Arc;
+
+pub use crate::host_fs::{Stderr, Stdin, Stdout};
+
+#[derive(Clone)]
+pub struct VfsFileSystem {
```

Comment:
What's the difference between a virtual file system file system (I'm just expandind the name here :-p), and an in-memory file system?